### PR TITLE
Feature/linux installer distribution

### PR DESCRIPTION
**Pull request by Minkiu**
## Changelog Description
Auto update logic for Linux, the launcher will now be able to fetch and unpack a new version of ayon based on the installers available in the server.

## Additional Notes
I might force push changes to this PR, if you clone it, you will need to do `git pull --rebase`

## Testing notes:
Same testing notes as in https://github.com/ynput/ayon-launcher/pull/9

### CODE REVIEW

File: common/ayon_common/distribution/control.py [hunk 1]
```diff
@@ -34,6 +34,7 @@
     DependencyItem,
     Bundle,
 )
+from .file_handler import RemoteFileHandler
```
[the remark this answers is not in the document]
Comment (iLLiCiTiT) [reply common/ayon_common/distribution/control.py:37]: I've actually just wanted to create PR which removes the `unzip` from `RemoteFileHandler`.

File: common/ayon_common/distribution/control.py [hunk 2]
```diff
@@ -626,13 +627,32 @@ def _install_windows(self, filepath):
...
+            RemoteFileHandler.unzip(filepath, destination_path=install_root)
+        except Exception as e:
+            self.log.error(e)
+            raise InstallerDistributionError(
+                "Install process failed without known reason."
+                " Try to install AYON manually."
+            )
+
+        installer_dir = os.path.basename(filepath).replace(".tar.gz", "")
+        executable = os.path.join(install_root, installer_dir, "ayon")
```
Comment (iLLiCiTiT) [inline common/ayon_common/distribution/control.py:652]:
```suggestion
        subdir = os.path.basename(filepath)[:-len(".tar.gz")]
        executable = os.path.join(install_root, subdir, "ayon")
```

File: common/ayon_common/distribution/control.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Minkiu) [reply common/ayon_common/distribution/control.py]: Shall i make `unzip` redirect to the `common_utils` ?

[the remark this answers is not in the document]
Comment (iLLiCiTiT) [reply common/ayon_common/distribution/control.py]: I've removed it here https://github.com/ynput/ayon-launcher/pull/17

**Pull request merged**